main.c

```diff
@@ -1334,7 +1349,7 @@ uploadsnap(const char * fname, const char * imgfmt, const char * region,
 static char *
 registerimage(const char * region, const char * snapshot, const char * name,
     const char * desc, const char * arch, int sriov, int ena,
-    const char * key_id, const char * key_secret)
+    const char * key_id, const char * key_secret, const char * diskimage)
```

Comment:
We're getting the file size in uploadvolume and uploadsnap.  Return the size from those functions and pass it into this one rather than statting the file a second time.

Reply:
How should this be done?

Reply:
Add a "imgsz" parameter to each of the functions?